homeassistant/helpers/config_validation.py

```diff
@@ -1769,10 +1771,32 @@ def STATE_CONDITION_SCHEMA(value: Any) -> dict[str, Any]:
     )
 )
 
+
+def _backward_compat_trigger_schema(value: Any | None) -> Any:
+    """Backward compatibility for trigger schemas."""
+
+    if not isinstance(value, dict):
+        return value
```

Comment:
Let's cover this.
We should add tests in tests/helpers/test_config_validation.py which call `TRIGGER_SCHEMA` and/or `_backward_compat_trigger_schema` with all the cases covered.